core/plugins/org.polarsys.capella.core.platform.sirius.ui.actions/src/org/polarsys/capella/core/platform/sirius/ui/handlers/DeleteHandler.java

```diff
@@ -36,10 +37,7 @@ public Object execute(ExecutionEvent event) throws ExecutionException {
 
     if (selection instanceof IStructuredSelection) {
       List<?> elementsToRemove = ((IStructuredSelection) selection).toList();
-      List<EObject> eObjectsToRemove = elementsToRemove.stream()//
-          .filter(EObject.class::isInstance) //
-          .map(EObject.class::cast).collect(Collectors.toList());
-
+      Collection<EObject> eObjectsToRemove = CapellaAdapterHelper.resolveDescriptorsOrBusinessObjects(elementsToRemove);
       if (!eObjectsToRemove.isEmpty()) {
 
         ExecutionManager executionManager = TransactionHelper.getExecutionManager(eObjectsToRemove);
```

Comment:
It should be `CapellaAdapterHelper.resolveBusinessObject`, we can not use `CapellaAdapterHelper.resolveDescriptorsOrBusinessObjects(eObjects)` since we want to keep `DRepresentationDescriptor` as they are, and not obtain their target (I guess if the user clicks on a descriptor we want to delete the descriptor, and not the semantic target).